src/modules/link.cpp

```diff
@@ -1,22 +1,123 @@
 //
 // Link Module
 //
-
+#include <Arduino.h>
+#include <FastCRC.h>
+#include "../pt/pt.h"
 #include "../modules/link.h"
 #include "../modules/module.h"
+#include "../hal/serial.h"
+
+FastCRC16 CRC16;  // this is the class to create crc16 to match the CCITT crc16 that rpi is using
 
-// Public Variables
-// TODO: Fix this variable name :(
+// Private Variables
 struct link comm;
+static struct pt serialThread;
+static struct pt serialReadThread;
+static struct pt serialSendThread;
+
+uint16_t calc_crc;
+
+// copy the public data packet to this buffer because there might be changes
+// by other modules while serial.cpp is waiting to send this
+data_packet_def update_crc_data_packet;
+
+#ifdef SERIAL_DEBUG
+extern HardwareSerial &serial_debug;
+#endif
+
+static PT_THREAD(serialReadThreadMain(struct pt* pt))
+{
+  PT_BEGIN(pt);
+
+  PT_WAIT_UNTIL(pt, serialHalGetData() != HAL_IN_PROGRESS);
+
+  if (watchdog_exceeded == true)
+  {
+    comm.public_command_packet.alarm_bits |= 1 << ALARM_DROPPED_PACKET;
```

Comment:
`ALARM_DROPPED_PACKET` doesnt define an offset, but the bit itself (which is a style I do like better), so this should be `alarm_bits |= ALARM_DROPPED_PACKET` instead of `alarm_bits |= 1 << ALARM_DROPPED_PACKET`

Reply:
I have fixed the action to match the bit pattern.